controllers/etcd_copy_backups_controller.go

```diff
@@ -0,0 +1,409 @@
+	}
+	if etcdCopyBackupsTask.Spec.MaxBackups != nil {
+		values["maxBackups"] = *etcdCopyBackupsTask.Spec.MaxBackups
+	}
+
+	if etcdCopyBackupsTask.Spec.Image != nil {
+		values["image"] = etcdCopyBackupsTask.Spec.Image
+	} else {
+		images, err := imagevector.FindImages(r.imageVector, []string{common.BackupRestore})
+		if err != nil {
+			return map[string]interface{}{}, err
+		}
+		val, ok := images[common.BackupRestore]
+		if !ok {
+			return nil, errors.New("Unable to find etcdbrctl image")
+		}
+		values["image"] = val.String()
+	}
+
+	return values, nil
+}
+
+func (r *EtcdCopyBackupsTaskReconciler) addFinalizerToObject(ctx context.Context, object client.Object, logger logr.Logger) error {
+	if !sets.NewString(object.GetFinalizers()...).Has(FinalizerName) {
+		objectCopy := object.DeepCopyObject().(client.Object)
+		finalizers := sets.NewString(objectCopy.GetFinalizers()...)
+		finalizers.Insert(FinalizerName)
+		objectCopy.SetFinalizers(finalizers.UnsortedList())
+		if err := r.Patch(ctx, objectCopy, client.MergeFrom(object)); client.IgnoreNotFound(err) != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *EtcdCopyBackupsTaskReconciler) removeFinalizerFromObject(ctx context.Context, object client.Object, logger logr.Logger) error {
+	if sets.NewString(object.GetFinalizers()...).Has(FinalizerName) {
+		objectCopy := object.DeepCopyObject().(client.Object)
+		finalizers := sets.NewString(objectCopy.GetFinalizers()...)
+		finalizers.Delete(FinalizerName)
+		objectCopy.SetFinalizers(finalizers.UnsortedList())
+		if err := r.Patch(ctx, objectCopy, client.MergeFrom(object)); client.IgnoreNotFound(err) != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *EtcdCopyBackupsTaskReconciler) checkForDuplicateOperation(ctx context.Context, etcdCopyBackupsTask *druidv1alpha1.EtcdCopyBackupsTask) error {
+	taskList := druidv1alpha1.EtcdCopyBackupsTaskList{}
+
+	if err := r.List(ctx, &taskList, client.InNamespace(etcdCopyBackupsTask.Namespace)); err != nil {
+		return err
+	}
+	for _, task := range taskList.Items {
+		if !reflect.DeepEqual(task.Name, etcdCopyBackupsTask.Name) &&
+			reflect.DeepEqual(task.Spec.TargetStore.Provider, etcdCopyBackupsTask.Spec.TargetStore.Provider) &&
+			reflect.DeepEqual(task.Spec.TargetStore.Container, etcdCopyBackupsTask.Spec.TargetStore.Container) &&
+			reflect.DeepEqual(task.Spec.TargetStore.Prefix, etcdCopyBackupsTask.Spec.TargetStore.Prefix) {
+			return fmt.Errorf("duplicate target configuration with task %s", task.Name)
+		}
+	}
+
+	return nil
+}
+
+// SetupWithManager sets up manager with a new controller and ec as the reconcile.Reconciler
+func (r *EtcdCopyBackupsTaskReconciler) SetupWithManager(mgr ctrl.Manager, ignoreOperationAnnotation bool) error {
+	builder := ctrl.NewControllerManagedBy(mgr)
+	builder = builder.WithEventFilter(buildPredicate(ignoreOperationAnnotation)).For(&druidv1alpha1.EtcdCopyBackupsTask{})
+	if ignoreOperationAnnotation {
+		builder = builder.Owns(&batchv1.Job{})
```

Comment:
Can you explain why`Job`s are only watched when `ignoreOperationAnnotation`?
Shouldn't we either decide to automatically enqueue after `30s` or to watch `Job`s, but not both?